command_line_assistant/utils/files.py

```diff
@@ -93,3 +100,94 @@ def write_file(contents: Union[str, bytes], path: Path, mode: int = 0o600) -> No
             path,
             str(e),
         )
+
+
+class NamedFileLock:
+    """File-based lock mechanism for general locking purposes.
+
+    This class provides a way to track state across processes
+    using a lock file. The lock file contains the PID of the capturing process.
+
+    Usage:
+        with NamedFileLock() as lock:
+            # Execute your operations
+        # Lock is automatically released
+    """
+
+    def __init__(self, name: str = "lock"):
```

Comment:
Maybe `name` does not need a default value. This forces the caller of `NamedFileLock` to provide a name. Seems reasonable to me.

Reply:
```suggestion
    def __init__(self, name: str):
```

That was a last minute addition I did. Was not sure either if having a default there was good or not.